src/DataFramesMeta.jl

```diff
@@ -387,17 +386,29 @@ function transform(d::Union{AbstractDataFrame, AbstractDict}; kwargs...)
     return result
 end
 
+
 function transform(g::GroupedDataFrame; kwargs...)
     result = DataFrame(g)
     idx2 = cumsum(Int[size(g[i],1) for i in 1:length(g)])
     idx1 = [1; 1 + idx2[1:end-1]]
+
+    function spread_scalar(x::Vector, length::Int)
+        return x 
+    end
+
+    function spread_scalar(x::CategoricalArrays.CategoricalValue, length::Int)
+        vec = CategoricalArray(fill(x, length))
+        levels!(vec, CategoricalArrays.index(CategoricalArrays.pool(x)))
```

Comment:
why this line is needed? I understand that without it `vec` has only one level? Yes?

Reply:
If we have a `CategoricalArray[1,2,1,2,1]` , I was thinking that if you take the first element of each group, filling in the rest of the group with 1 value, we still want to have the new vector have the same levels as the original one. 

Without that line you also get this weird case where the `pool` of the array and the `pool` of the elements are different.

Reply:
This is what I thought. Just wanted to be sure.